api/renterd/server.go

```diff
@@ -177,12 +180,95 @@ func (s *server) contractsRenewHandler(w http.ResponseWriter, req *http.Request,
 
 	// TODO: compare request settings to current host settings
 
-	contract, parent, err := session.RenewContract(crr.RenterKey, crr.Contract, crr.RenterFunds, crr.HostCollateral, crr.Extension)
+	contract, parent, err := session.RenewContract(rrr.RenterKey, rrr.Contract, rrr.RenterFunds, rrr.HostCollateral, rrr.Extension, s.w, s.tp)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-	api.WriteJSON(w, ContractsRenewResponse{contract, parent})
+	api.WriteJSON(w, RHPRenewResponse{contract, parent})
+}
+
+func (s *server) rhpReadHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var rrr RHPReadRequest
+	if err := json.NewDecoder(req.Body).Decode(&rrr); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	session, err := renter.NewSession(rrr.NetAddress, rrr.HostKey, s.cm)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer session.Close()
+	if err := session.Lock(req.Context(), rrr.ContractID, rrr.RenterKey); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	var refundAccountID types.PublicKey // TODO: ???
```

Comment:
@n8maninger what pubkey should we be using for these?

Reply:
It's an EA account; any key pair will work. I've been using `renterKey.Public()` for simplicity while testing; it's probably OK not to expose anything more advanced in renterd.